upstream-sha256: d1e4dee4067dff336fa390796b147810738fd7e0785abb0f32ac1a1dbecca022
actual-sha256: d0c31ed9594808e917b1e807bebfe809bf8946fb47a7e5ba48301d9d561861c3
--- upstream.expected
+++ tolk.actual
@@ -3,199 +3,207 @@
    2 Ops.CAST            :            dtypes.int                               ['32']                           dtypes.int
    3 Ops.SPECIAL         :            dtypes.int                               [2]                              lidx0
    4 Ops.INDEX           :            dtypes.float                             [0, 3]                           None
-   5 Ops.PARAM           :            dtypes.bfloat16                          []                               ParamArg(1, dtypes.bfloat16, 256, device='CPU')
-   6 Ops.CONST           :            dtypes.weakint                           []                               15
-   7 Ops.CAST            :            dtypes.int                               ['15']                           dtypes.int
-   8 Ops.AND             :            dtypes.int                               [3, 7]                           None
-   9 Ops.CONST           :            dtypes.weakint                           []                               4
-  10 Ops.CAST            :            dtypes.int                               ['4']                            dtypes.int
-  11 Ops.SHL             :            dtypes.int                               [8, 10]                          None
-  12 Ops.INDEX           :            dtypes.bfloat16                          [5, 11]                          None
-  13 Ops.LOAD            :            dtypes.bfloat16                          [12]                             None
-  14 Ops.CONST           :            dtypes.weakint                           []                               1
-  15 Ops.CAST            :            dtypes.int                               ['1']                            dtypes.int
-  16 Ops.ADD             :            dtypes.int                               [11, 15]                         None
-  17 Ops.INDEX           :            dtypes.bfloat16                          [5, 16]                          None
-  18 Ops.LOAD            :            dtypes.bfloat16                          [17]                             None
-  19 Ops.CONST           :            dtypes.weakint                           []                               2
-  20 Ops.CAST            :            dtypes.int                               ['2']                            dtypes.int
-  21 Ops.ADD             :            dtypes.int                               [11, 20]                         None
-  22 Ops.INDEX           :            dtypes.bfloat16                          [5, 21]                          None
-  23 Ops.LOAD            :            dtypes.bfloat16                          [22]                             None
-  24 Ops.CONST           :            dtypes.weakint                           []                               3
-  25 Ops.CAST            :            dtypes.int                               ['3']                            dtypes.int
-  26 Ops.ADD             :            dtypes.int                               [11, 25]                         None
-  27 Ops.INDEX           :            dtypes.bfloat16                          [5, 26]                          None
-  28 Ops.LOAD            :            dtypes.bfloat16                          [27]                             None
-  29 Ops.ADD             :            dtypes.int                               [11, 10]                         None
-  30 Ops.INDEX           :            dtypes.bfloat16                          [5, 29]                          None
-  31 Ops.LOAD            :            dtypes.bfloat16                          [30]                             None
-  32 Ops.CONST           :            dtypes.weakint                           []                               5
-  33 Ops.CAST            :            dtypes.int                               ['5']                            dtypes.int
-  34 Ops.ADD             :            dtypes.int                               [11, 33]                         None
-  35 Ops.INDEX           :            dtypes.bfloat16                          [5, 34]                          None
-  36 Ops.LOAD            :            dtypes.bfloat16                          [35]                             None
-  37 Ops.CONST           :            dtypes.weakint                           []                               6
-  38 Ops.CAST            :            dtypes.int                               ['6']                            dtypes.int
-  39 Ops.ADD             :            dtypes.int                               [11, 38]                         None
-  40 Ops.INDEX           :            dtypes.bfloat16                          [5, 39]                          None
-  41 Ops.LOAD            :            dtypes.bfloat16                          [40]                             None
-  42 Ops.CONST           :            dtypes.weakint                           []                               7
-  43 Ops.CAST            :            dtypes.int                               ['7']                            dtypes.int
-  44 Ops.ADD             :            dtypes.int                               [11, 43]                         None
-  45 Ops.INDEX           :            dtypes.bfloat16                          [5, 44]                          None
-  46 Ops.LOAD            :            dtypes.bfloat16                          [45]                             None
-  47 Ops.CONST           :            dtypes.weakint                           []                               8
-  48 Ops.CAST            :            dtypes.int                               ['8']                            dtypes.int
-  49 Ops.ADD             :            dtypes.int                               [11, 48]                         None
-  50 Ops.INDEX           :            dtypes.bfloat16                          [5, 49]                          None
-  51 Ops.LOAD            :            dtypes.bfloat16                          [50]                             None
-  52 Ops.CONST           :            dtypes.weakint                           []                               9
-  53 Ops.CAST            :            dtypes.int                               ['9']                            dtypes.int
-  54 Ops.ADD             :            dtypes.int                               [11, 53]                         None
-  55 Ops.INDEX           :            dtypes.bfloat16                          [5, 54]                          None
-  56 Ops.LOAD            :            dtypes.bfloat16                          [55]                             None
-  57 Ops.CONST           :            dtypes.weakint                           []                               10
-  58 Ops.CAST            :            dtypes.int                               ['10']                           dtypes.int
-  59 Ops.ADD             :            dtypes.int                               [11, 58]                         None
-  60 Ops.INDEX           :            dtypes.bfloat16                          [5, 59]                          None
-  61 Ops.LOAD            :            dtypes.bfloat16                          [60]                             None
-  62 Ops.CONST           :            dtypes.weakint                           []                               11
-  63 Ops.CAST            :            dtypes.int                               ['11']                           dtypes.int
-  64 Ops.ADD             :            dtypes.int                               [11, 63]                         None
-  65 Ops.INDEX           :            dtypes.bfloat16                          [5, 64]                          None
-  66 Ops.LOAD            :            dtypes.bfloat16                          [65]                             None
-  67 Ops.CONST           :            dtypes.weakint                           []                               12
-  68 Ops.CAST            :            dtypes.int                               ['12']                           dtypes.int
-  69 Ops.ADD             :            dtypes.int                               [11, 68]                         None
-  70 Ops.INDEX           :            dtypes.bfloat16                          [5, 69]                          None
-  71 Ops.LOAD            :            dtypes.bfloat16                          [70]                             None
-  72 Ops.CONST           :            dtypes.weakint                           []                               13
-  73 Ops.CAST            :            dtypes.int                               ['13']                           dtypes.int
-  74 Ops.ADD             :            dtypes.int                               [11, 73]                         None
-  75 Ops.INDEX           :            dtypes.bfloat16                          [5, 74]                          None
-  76 Ops.LOAD            :            dtypes.bfloat16                          [75]                             None
-  77 Ops.CONST           :            dtypes.weakint                           []                               14
-  78 Ops.CAST            :            dtypes.int                               ['14']                           dtypes.int
-  79 Ops.ADD             :            dtypes.int                               [11, 78]                         None
-  80 Ops.INDEX           :            dtypes.bfloat16                          [5, 79]                          None
-  81 Ops.LOAD            :            dtypes.bfloat16                          [80]                             None
-  82 Ops.ADD             :            dtypes.int                               [11, 7]                          None
-  83 Ops.INDEX           :            dtypes.bfloat16                          [5, 82]                          None
-  84 Ops.LOAD            :            dtypes.bfloat16                          [83]                             None
-  85 Ops.STACK           :            dtypes.bfloat16                          [13, 18, 23, 28, 31, 36, 41, 46, 51, 56, 61, 66, 71, 76, 81, 84] None
-  86 Ops.PARAM           :            dtypes.bfloat16                          []                               ParamArg(2, dtypes.bfloat16, 256, device='CPU')
-  87 Ops.INDEX           :            dtypes.bfloat16                          [86, 8]                          None
-  88 Ops.LOAD            :            dtypes.bfloat16                          [87]                             None
-  89 Ops.CONST           :            dtypes.weakint                           []                               16
-  90 Ops.CAST            :            dtypes.int                               ['16']                           dtypes.int
-  91 Ops.ADD             :            dtypes.int                               [8, 90]                          None
-  92 Ops.INDEX           :            dtypes.bfloat16                          [86, 91]                         None
-  93 Ops.LOAD            :            dtypes.bfloat16                          [92]                             None
-  94 Ops.ADD             :            dtypes.int                               [8, 2]                           None
-  95 Ops.INDEX           :            dtypes.bfloat16                          [86, 94]                         None
-  96 Ops.LOAD            :            dtypes.bfloat16                          [95]                             None
-  97 Ops.CONST           :            dtypes.weakint                           []                               48
-  98 Ops.CAST            :            dtypes.int                               ['48']                           dtypes.int
-  99 Ops.ADD             :            dtypes.int                               [8, 98]                          None
- 100 Ops.INDEX           :            dtypes.bfloat16                          [86, 99]                         None
- 101 Ops.LOAD            :            dtypes.bfloat16                          [100]                            None
- 102 Ops.CONST           :            dtypes.weakint                           []                               64
- 103 Ops.CAST            :            dtypes.int                               ['64']                           dtypes.int
- 104 Ops.ADD             :            dtypes.int                               [8, 103]                         None
- 105 Ops.INDEX           :            dtypes.bfloat16                          [86, 104]                        None
- 106 Ops.LOAD            :            dtypes.bfloat16                          [105]                            None
- 107 Ops.CONST           :            dtypes.weakint                           []                               80
- 108 Ops.CAST            :            dtypes.int                               ['80']                           dtypes.int
- 109 Ops.ADD             :            dtypes.int                               [8, 108]                         None
- 110 Ops.INDEX           :            dtypes.bfloat16                          [86, 109]                        None
- 111 Ops.LOAD            :            dtypes.bfloat16                          [110]                            None
- 112 Ops.CONST           :            dtypes.weakint                           []                               96
- 113 Ops.CAST            :            dtypes.int                               ['96']                           dtypes.int
- 114 Ops.ADD             :            dtypes.int                               [8, 113]                         None
- 115 Ops.INDEX           :            dtypes.bfloat16                          [86, 114]                        None
- 116 Ops.LOAD            :            dtypes.bfloat16                          [115]                            None
- 117 Ops.CONST           :            dtypes.weakint                           []                               112
- 118 Ops.CAST            :            dtypes.int                               ['112']                          dtypes.int
- 119 Ops.ADD             :            dtypes.int                               [8, 118]                         None
- 120 Ops.INDEX           :            dtypes.bfloat16                          [86, 119]                        None
- 121 Ops.LOAD            :            dtypes.bfloat16                          [120]                            None
- 122 Ops.CONST           :            dtypes.weakint                           []                               128
- 123 Ops.CAST            :            dtypes.int                               ['128']                          dtypes.int
- 124 Ops.ADD             :            dtypes.int                               [8, 123]                         None
- 125 Ops.INDEX           :            dtypes.bfloat16                          [86, 124]                        None
- 126 Ops.LOAD            :            dtypes.bfloat16                          [125]                            None
- 127 Ops.CONST           :            dtypes.weakint                           []                               144
- 128 Ops.CAST            :            dtypes.int                               ['144']                          dtypes.int
- 129 Ops.ADD             :            dtypes.int                               [8, 128]                         None
- 130 Ops.INDEX           :            dtypes.bfloat16                          [86, 129]                        None
- 131 Ops.LOAD            :            dtypes.bfloat16                          [130]                            None
- 132 Ops.CONST           :            dtypes.weakint                           []                               160
- 133 Ops.CAST            :            dtypes.int                               ['160']                          dtypes.int
- 134 Ops.ADD             :            dtypes.int                               [8, 133]                         None
- 135 Ops.INDEX           :            dtypes.bfloat16                          [86, 134]                        None
- 136 Ops.LOAD            :            dtypes.bfloat16                          [135]                            None
- 137 Ops.CONST           :            dtypes.weakint                           []                               176
- 138 Ops.CAST            :            dtypes.int                               ['176']                          dtypes.int
- 139 Ops.ADD             :            dtypes.int                               [8, 138]                         None
- 140 Ops.INDEX           :            dtypes.bfloat16                          [86, 139]                        None
- 141 Ops.LOAD            :            dtypes.bfloat16                          [140]                            None
- 142 Ops.CONST           :            dtypes.weakint                           []                               192
- 143 Ops.CAST            :            dtypes.int                               ['192']                          dtypes.int
- 144 Ops.ADD             :            dtypes.int                               [8, 143]                         None
- 145 Ops.INDEX           :            dtypes.bfloat16                          [86, 144]                        None
- 146 Ops.LOAD            :            dtypes.bfloat16                          [145]                            None
- 147 Ops.CONST           :            dtypes.weakint                           []                               208
- 148 Ops.CAST            :            dtypes.int                               ['208']                          dtypes.int
- 149 Ops.ADD             :            dtypes.int                               [8, 148]                         None
- 150 Ops.INDEX           :            dtypes.bfloat16                          [86, 149]                        None
- 151 Ops.LOAD            :            dtypes.bfloat16                          [150]                            None
- 152 Ops.CONST           :            dtypes.weakint                           []                               224
- 153 Ops.CAST            :            dtypes.int                               ['224']                          dtypes.int
- 154 Ops.ADD             :            dtypes.int                               [8, 153]                         None
- 155 Ops.INDEX           :            dtypes.bfloat16                          [86, 154]                        None
- 156 Ops.LOAD            :            dtypes.bfloat16                          [155]                            None
- 157 Ops.CONST           :            dtypes.weakint                           []                               240
- 158 Ops.CAST            :            dtypes.int                               ['240']                          dtypes.int
- 159 Ops.ADD             :            dtypes.int                               [8, 158]                         None
- 160 Ops.INDEX           :            dtypes.bfloat16                          [86, 159]                        None
- 161 Ops.LOAD            :            dtypes.bfloat16                          [160]                            None
- 162 Ops.STACK           :            dtypes.bfloat16                          [88, 93, 96, 101, 106, 111, 116, 121, 126, 131, 136, 141, 146, 151, 156, 161] None
- 163 Ops.CONST           :            dtypes.weakfloat                         []                               0.0
- 164 Ops.CAST            :            dtypes.float                             ['0.0']                          dtypes.float
- 165 Ops.STACK           :            dtypes.float                             [164, 164, 164, 164, 164, 164, 164, 164] None
- 166 Ops.WMMA            :            dtypes.float                             [85, 162, 165]                   ((16, 16, 16), dtypes.bfloat16, 32, None)
+   5 Ops.CONST           :            dtypes.weakfloat                         []                               0.0
+   6 Ops.CAST            :            dtypes.float                             ['0.0']                          dtypes.float
+   7 Ops.PARAM           :            dtypes.bfloat16                          []                               ParamArg(1, dtypes.bfloat16, 256, device='CPU')
+   8 Ops.CONST           :            dtypes.weakint                           []                               15
+   9 Ops.CAST            :            dtypes.int                               ['15']                           dtypes.int
+  10 Ops.AND             :            dtypes.int                               [3, 9]                           None
+  11 Ops.CONST           :            dtypes.weakint                           []                               4
+  12 Ops.CAST            :            dtypes.int                               ['4']                            dtypes.int
+  13 Ops.SHL             :            dtypes.int                               [10, 12]                         None
+  14 Ops.INDEX           :            dtypes.bfloat16                          [7, 13]                          None
+  15 Ops.LOAD            :            dtypes.bfloat16                          [14]                             None
+  16 Ops.CONST           :            dtypes.weakint                           []                               1
+  17 Ops.CAST            :            dtypes.int                               ['1']                            dtypes.int
+  18 Ops.ADD             :            dtypes.int                               [13, 17]                         None
+  19 Ops.INDEX           :            dtypes.bfloat16                          [7, 18]                          None
+  20 Ops.LOAD            :            dtypes.bfloat16                          [19]                             None
+  21 Ops.CONST           :            dtypes.weakint                           []                               2
+  22 Ops.CAST            :            dtypes.int                               ['2']                            dtypes.int
+  23 Ops.ADD             :            dtypes.int                               [13, 22]                         None
+  24 Ops.INDEX           :            dtypes.bfloat16                          [7, 23]                          None
+  25 Ops.LOAD            :            dtypes.bfloat16                          [24]                             None
+  26 Ops.CONST           :            dtypes.weakint                           []                               3
+  27 Ops.CAST            :            dtypes.int                               ['3']                            dtypes.int
+  28 Ops.ADD             :            dtypes.int                               [13, 27]                         None
+  29 Ops.INDEX           :            dtypes.bfloat16                          [7, 28]                          None
+  30 Ops.LOAD            :            dtypes.bfloat16                          [29]                             None
+  31 Ops.ADD             :            dtypes.int                               [13, 12]                         None
+  32 Ops.INDEX           :            dtypes.bfloat16                          [7, 31]                          None
+  33 Ops.LOAD            :            dtypes.bfloat16                          [32]                             None
+  34 Ops.CONST           :            dtypes.weakint                           []                               5
+  35 Ops.CAST            :            dtypes.int                               ['5']                            dtypes.int
+  36 Ops.ADD             :            dtypes.int                               [13, 35]                         None
+  37 Ops.INDEX           :            dtypes.bfloat16                          [7, 36]                          None
+  38 Ops.LOAD            :            dtypes.bfloat16                          [37]                             None
+  39 Ops.CONST           :            dtypes.weakint                           []                               6
+  40 Ops.CAST            :            dtypes.int                               ['6']                            dtypes.int
+  41 Ops.ADD             :            dtypes.int                               [13, 40]                         None
+  42 Ops.INDEX           :            dtypes.bfloat16                          [7, 41]                          None
+  43 Ops.LOAD            :            dtypes.bfloat16                          [42]                             None
+  44 Ops.CONST           :            dtypes.weakint                           []                               7
+  45 Ops.CAST            :            dtypes.int                               ['7']                            dtypes.int
+  46 Ops.ADD             :            dtypes.int                               [13, 45]                         None
+  47 Ops.INDEX           :            dtypes.bfloat16                          [7, 46]                          None
+  48 Ops.LOAD            :            dtypes.bfloat16                          [47]                             None
+  49 Ops.CONST           :            dtypes.weakint                           []                               8
+  50 Ops.CAST            :            dtypes.int                               ['8']                            dtypes.int
+  51 Ops.ADD             :            dtypes.int                               [13, 50]                         None
+  52 Ops.INDEX           :            dtypes.bfloat16                          [7, 51]                          None
+  53 Ops.LOAD            :            dtypes.bfloat16                          [52]                             None
+  54 Ops.CONST           :            dtypes.weakint                           []                               9
+  55 Ops.CAST            :            dtypes.int                               ['9']                            dtypes.int
+  56 Ops.ADD             :            dtypes.int                               [13, 55]                         None
+  57 Ops.INDEX           :            dtypes.bfloat16                          [7, 56]                          None
+  58 Ops.LOAD            :            dtypes.bfloat16                          [57]                             None
+  59 Ops.CONST           :            dtypes.weakint                           []                               10
+  60 Ops.CAST            :            dtypes.int                               ['10']                           dtypes.int
+  61 Ops.ADD             :            dtypes.int                               [13, 60]                         None
+  62 Ops.INDEX           :            dtypes.bfloat16                          [7, 61]                          None
+  63 Ops.LOAD            :            dtypes.bfloat16                          [62]                             None
+  64 Ops.CONST           :            dtypes.weakint                           []                               11
+  65 Ops.CAST            :            dtypes.int                               ['11']                           dtypes.int
+  66 Ops.ADD             :            dtypes.int                               [13, 65]                         None
+  67 Ops.INDEX           :            dtypes.bfloat16                          [7, 66]                          None
+  68 Ops.LOAD            :            dtypes.bfloat16                          [67]                             None
+  69 Ops.CONST           :            dtypes.weakint                           []                               12
+  70 Ops.CAST            :            dtypes.int                               ['12']                           dtypes.int
+  71 Ops.ADD             :            dtypes.int                               [13, 70]                         None
+  72 Ops.INDEX           :            dtypes.bfloat16                          [7, 71]                          None
+  73 Ops.LOAD            :            dtypes.bfloat16                          [72]                             None
+  74 Ops.CONST           :            dtypes.weakint                           []                               13
+  75 Ops.CAST            :            dtypes.int                               ['13']                           dtypes.int
+  76 Ops.ADD             :            dtypes.int                               [13, 75]                         None
+  77 Ops.INDEX           :            dtypes.bfloat16                          [7, 76]                          None
+  78 Ops.LOAD            :            dtypes.bfloat16                          [77]                             None
+  79 Ops.CONST           :            dtypes.weakint                           []                               14
+  80 Ops.CAST            :            dtypes.int                               ['14']                           dtypes.int
+  81 Ops.ADD             :            dtypes.int                               [13, 80]                         None
+  82 Ops.INDEX           :            dtypes.bfloat16                          [7, 81]                          None
+  83 Ops.LOAD            :            dtypes.bfloat16                          [82]                             None
+  84 Ops.ADD             :            dtypes.int                               [13, 9]                          None
+  85 Ops.INDEX           :            dtypes.bfloat16                          [7, 84]                          None
+  86 Ops.LOAD            :            dtypes.bfloat16                          [85]                             None
+  87 Ops.STACK           :            dtypes.bfloat16                          [15, 20, 25, 30, 33, 38, 43, 48, 53, 58, 63, 68, 73, 78, 83, 86] None
+  88 Ops.PARAM           :            dtypes.bfloat16                          []                               ParamArg(2, dtypes.bfloat16, 256, device='CPU')
+  89 Ops.INDEX           :            dtypes.bfloat16                          [88, 10]                         None
+  90 Ops.LOAD            :            dtypes.bfloat16                          [89]                             None
+  91 Ops.CONST           :            dtypes.weakint                           []                               16
+  92 Ops.CAST            :            dtypes.int                               ['16']                           dtypes.int
+  93 Ops.ADD             :            dtypes.int                               [10, 92]                         None
+  94 Ops.INDEX           :            dtypes.bfloat16                          [88, 93]                         None
+  95 Ops.LOAD            :            dtypes.bfloat16                          [94]                             None
+  96 Ops.ADD             :            dtypes.int                               [10, 2]                          None
+  97 Ops.INDEX           :            dtypes.bfloat16                          [88, 96]                         None
+  98 Ops.LOAD            :            dtypes.bfloat16                          [97]                             None
+  99 Ops.CONST           :            dtypes.weakint                           []                               48
+ 100 Ops.CAST            :            dtypes.int                               ['48']                           dtypes.int
+ 101 Ops.ADD             :            dtypes.int                               [10, 100]                        None
+ 102 Ops.INDEX           :            dtypes.bfloat16                          [88, 101]                        None
+ 103 Ops.LOAD            :            dtypes.bfloat16                          [102]                            None
+ 104 Ops.CONST           :            dtypes.weakint                           []                               64
+ 105 Ops.CAST            :            dtypes.int                               ['64']                           dtypes.int
+ 106 Ops.ADD             :            dtypes.int                               [10, 105]                        None
+ 107 Ops.INDEX           :            dtypes.bfloat16                          [88, 106]                        None
+ 108 Ops.LOAD            :            dtypes.bfloat16                          [107]                            None
+ 109 Ops.CONST           :            dtypes.weakint                           []                               80
+ 110 Ops.CAST            :            dtypes.int                               ['80']                           dtypes.int
+ 111 Ops.ADD             :            dtypes.int                               [10, 110]                        None
+ 112 Ops.INDEX           :            dtypes.bfloat16                          [88, 111]                        None
+ 113 Ops.LOAD            :            dtypes.bfloat16                          [112]                            None
+ 114 Ops.CONST           :            dtypes.weakint                           []                               96
+ 115 Ops.CAST            :            dtypes.int                               ['96']                           dtypes.int
+ 116 Ops.ADD             :            dtypes.int                               [10, 115]                        None
+ 117 Ops.INDEX           :            dtypes.bfloat16                          [88, 116]                        None
+ 118 Ops.LOAD            :            dtypes.bfloat16                          [117]                            None
+ 119 Ops.CONST           :            dtypes.weakint                           []                               112
+ 120 Ops.CAST            :            dtypes.int                               ['112']                          dtypes.int
+ 121 Ops.ADD             :            dtypes.int                               [10, 120]                        None
+ 122 Ops.INDEX           :            dtypes.bfloat16                          [88, 121]                        None
+ 123 Ops.LOAD            :            dtypes.bfloat16                          [122]                            None
+ 124 Ops.CONST           :            dtypes.weakint                           []                               128
+ 125 Ops.CAST            :            dtypes.int                               ['128']                          dtypes.int
+ 126 Ops.ADD             :            dtypes.int                               [10, 125]                        None
+ 127 Ops.INDEX           :            dtypes.bfloat16                          [88, 126]                        None
+ 128 Ops.LOAD            :            dtypes.bfloat16                          [127]                            None
+ 129 Ops.CONST           :            dtypes.weakint                           []                               144
+ 130 Ops.CAST            :            dtypes.int                               ['144']                          dtypes.int
+ 131 Ops.ADD             :            dtypes.int                               [10, 130]                        None
+ 132 Ops.INDEX           :            dtypes.bfloat16                          [88, 131]                        None
+ 133 Ops.LOAD            :            dtypes.bfloat16                          [132]                            None
+ 134 Ops.CONST           :            dtypes.weakint                           []                               160
+ 135 Ops.CAST            :            dtypes.int                               ['160']                          dtypes.int
+ 136 Ops.ADD             :            dtypes.int                               [10, 135]                        None
+ 137 Ops.INDEX           :            dtypes.bfloat16                          [88, 136]                        None
+ 138 Ops.LOAD            :            dtypes.bfloat16                          [137]                            None
+ 139 Ops.CONST           :            dtypes.weakint                           []                               176
+ 140 Ops.CAST            :            dtypes.int                               ['176']                          dtypes.int
+ 141 Ops.ADD             :            dtypes.int                               [10, 140]                        None
+ 142 Ops.INDEX           :            dtypes.bfloat16                          [88, 141]                        None
+ 143 Ops.LOAD            :            dtypes.bfloat16                          [142]                            None
+ 144 Ops.CONST           :            dtypes.weakint                           []                               192
+ 145 Ops.CAST            :            dtypes.int                               ['192']                          dtypes.int
+ 146 Ops.ADD             :            dtypes.int                               [10, 145]                        None
+ 147 Ops.INDEX           :            dtypes.bfloat16                          [88, 146]                        None
+ 148 Ops.LOAD            :            dtypes.bfloat16                          [147]                            None
+ 149 Ops.CONST           :            dtypes.weakint                           []                               208
+ 150 Ops.CAST            :            dtypes.int                               ['208']                          dtypes.int
+ 151 Ops.ADD             :            dtypes.int                               [10, 150]                        None
+ 152 Ops.INDEX           :            dtypes.bfloat16                          [88, 151]                        None
+ 153 Ops.LOAD            :            dtypes.bfloat16                          [152]                            None
+ 154 Ops.CONST           :            dtypes.weakint                           []                               224
+ 155 Ops.CAST            :            dtypes.int                               ['224']                          dtypes.int
+ 156 Ops.ADD             :            dtypes.int                               [10, 155]                        None
+ 157 Ops.INDEX           :            dtypes.bfloat16                          [88, 156]                        None
+ 158 Ops.LOAD            :            dtypes.bfloat16                          [157]                            None
+ 159 Ops.CONST           :            dtypes.weakint                           []                               240
+ 160 Ops.CAST            :            dtypes.int                               ['240']                          dtypes.int
+ 161 Ops.ADD             :            dtypes.int                               [10, 160]                        None
+ 162 Ops.INDEX           :            dtypes.bfloat16                          [88, 161]                        None
+ 163 Ops.LOAD            :            dtypes.bfloat16                          [162]                            None
+ 164 Ops.STACK           :            dtypes.bfloat16                          [90, 95, 98, 103, 108, 113, 118, 123, 128, 133, 138, 143, 148, 153, 158, 163] None
+ 165 Ops.STACK           :            dtypes.float                             [6, 6, 6, 6, 6, 6, 6, 6]         None
+ 166 Ops.WMMA            :            dtypes.float                             [87, 164, 165]                   ((16, 16, 16), dtypes.bfloat16, 32, None)
  167 Ops.CONST           :            dtypes.weakint                           []                               0
  168 Ops.CAST            :            dtypes.int                               ['0']                            dtypes.int
  169 Ops.INDEX           :            dtypes.float                             [166, 168]                       None
- 170 Ops.STORE           :            dtypes.void                              [4, 169]                         None
- 171 Ops.ADD             :            dtypes.int                               [3, 2]                           None
- 172 Ops.INDEX           :            dtypes.float                             [0, 171]                         None
- 173 Ops.INDEX           :            dtypes.float                             [166, 15]                        None
- 174 Ops.STORE           :            dtypes.void                              [172, 173]                       None
- 175 Ops.ADD             :            dtypes.int                               [3, 123]                         None
- 176 Ops.INDEX           :            dtypes.float                             [0, 175]                         None
- 177 Ops.INDEX           :            dtypes.float                             [166, 10]                        None
- 178 Ops.STORE           :            dtypes.void                              [176, 177]                       None
- 179 Ops.ADD             :            dtypes.int                               [3, 133]                         None
- 180 Ops.INDEX           :            dtypes.float                             [0, 179]                         None
- 181 Ops.INDEX           :            dtypes.float                             [166, 33]                        None
- 182 Ops.STORE           :            dtypes.void                              [180, 181]                       None
- 183 Ops.ADD             :            dtypes.int                               [3, 103]                         None
- 184 Ops.INDEX           :            dtypes.float                             [0, 183]                         None
- 185 Ops.INDEX           :            dtypes.float                             [166, 20]                        None
- 186 Ops.STORE           :            dtypes.void                              [184, 185]                       None
- 187 Ops.ADD             :            dtypes.int                               [3, 113]                         None
+ 170 Ops.ADD             :            dtypes.float                             [6, 169]                         None
+ 171 Ops.STORE           :            dtypes.void                              [4, 170]                         None
+ 172 Ops.ADD             :            dtypes.int                               [3, 2]                           None
+ 173 Ops.INDEX           :            dtypes.float                             [0, 172]                         None
+ 174 Ops.INDEX           :            dtypes.float                             [166, 17]                        None
+ 175 Ops.ADD             :            dtypes.float                             [6, 174]                         None
+ 176 Ops.STORE           :            dtypes.void                              [173, 175]                       None
+ 177 Ops.ADD             :            dtypes.int                               [3, 125]                         None
+ 178 Ops.INDEX           :            dtypes.float                             [0, 177]                         None
+ 179 Ops.INDEX           :            dtypes.float                             [166, 12]                        None
+ 180 Ops.ADD             :            dtypes.float                             [6, 179]                         None
+ 181 Ops.STORE           :            dtypes.void                              [178, 180]                       None
+ 182 Ops.ADD             :            dtypes.int                               [3, 135]                         None
+ 183 Ops.INDEX           :            dtypes.float                             [0, 182]                         None
+ 184 Ops.INDEX           :            dtypes.float                             [166, 35]                        None
+ 185 Ops.ADD             :            dtypes.float                             [6, 184]                         None
+ 186 Ops.STORE           :            dtypes.void                              [183, 185]                       None
+ 187 Ops.ADD             :            dtypes.int                               [3, 105]                         None
  188 Ops.INDEX           :            dtypes.float                             [0, 187]                         None
- 189 Ops.INDEX           :            dtypes.float                             [166, 25]                        None
- 190 Ops.STORE           :            dtypes.void                              [188, 189]                       None
- 191 Ops.ADD             :            dtypes.int                               [3, 143]                         None
- 192 Ops.INDEX           :            dtypes.float                             [0, 191]                         None
- 193 Ops.INDEX           :            dtypes.float                             [166, 38]                        None
- 194 Ops.STORE           :            dtypes.void                              [192, 193]                       None
- 195 Ops.ADD             :            dtypes.int                               [3, 153]                         None
- 196 Ops.INDEX           :            dtypes.float                             [0, 195]                         None
- 197 Ops.INDEX           :            dtypes.float                             [166, 43]                        None
- 198 Ops.STORE           :            dtypes.void                              [196, 197]                       None
- 199 Ops.GROUP           :            dtypes.void                              [170, 174, 178, 182, 186, 190, 194, 198] None
- 200 Ops.SINK            :            dtypes.void                              [199]                            KernelInfo(name='r_32_2_2_2_2', applied_opts=(Opt(op=OptOps.TC, axis=0, arg=(-1, 0, 1)),), opts_to_apply=None, estimates=None, beam=0)
+ 189 Ops.INDEX           :            dtypes.float                             [166, 22]                        None
+ 190 Ops.ADD             :            dtypes.float                             [6, 189]                         None
+ 191 Ops.STORE           :            dtypes.void                              [188, 190]                       None
+ 192 Ops.ADD             :            dtypes.int                               [3, 115]                         None
+ 193 Ops.INDEX           :            dtypes.float                             [0, 192]                         None
+ 194 Ops.INDEX           :            dtypes.float                             [166, 27]                        None
+ 195 Ops.ADD             :            dtypes.float                             [6, 194]                         None
+ 196 Ops.STORE           :            dtypes.void                              [193, 195]                       None
+ 197 Ops.ADD             :            dtypes.int                               [3, 145]                         None
+ 198 Ops.INDEX           :            dtypes.float                             [0, 197]                         None
+ 199 Ops.INDEX           :            dtypes.float                             [166, 40]                        None
+ 200 Ops.ADD             :            dtypes.float                             [6, 199]                         None
+ 201 Ops.STORE           :            dtypes.void                              [198, 200]                       None
+ 202 Ops.ADD             :            dtypes.int                               [3, 155]                         None
+ 203 Ops.INDEX           :            dtypes.float                             [0, 202]                         None
+ 204 Ops.INDEX           :            dtypes.float                             [166, 45]                        None
+ 205 Ops.ADD             :            dtypes.float                             [6, 204]                         None
+ 206 Ops.STORE           :            dtypes.void                              [203, 205]                       None
+ 207 Ops.GROUP           :            dtypes.void                              [171, 176, 181, 186, 191, 196, 201, 206] None
+ 208 Ops.SINK            :            dtypes.void                              [207]                            KernelInfo(name='r_32_2_2_2_2', applied_opts=(Opt(op=OptOps.TC, axis=0, arg=(-1, 0, 1)),), opts_to_apply=None, estimates=None, beam=0)
